projects/js-packages/ai-client/src/data-flow/use-ai-context.ts

```diff
@@ -60,6 +60,20 @@ export default function useAiContext( {
 		onError?.( event?.detail );
 	}, [] );
 
+	/*
+	 * Trigger error handling for errors
+	 * that happened before the event source was created.
+	 */
+	useEffect( () => {
+		if ( ! requestingError ) {
+			return;
+		}
+
+		if ( onError ) {
+			onError( requestingError );
+		}
+	}, [ requestingError ] );
```

Comment:
imho, we should avoid as much as possible propagating the error through an effect hook if we could rely on listening events, better.

Reply:
Moved to the event approach that we discussed.